platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/MarkerViewManager.java

```diff
@@ -349,6 +349,9 @@ public void invalidateViewMarkersInBounds() {
                         if (adaptedView != null) {
 
                             // tilt
+                            if (marker.isFlat() && marker.getTilt() == 0) {
+                                marker.setTilt((float) mapboxMap.getCameraPosition().tilt);
+                            }
```

Comment:
looking into this again, the `getCameraPosiition` will in most cases result in a JNI call, for high trafficking code as `MarkerViewManager.update()` this might not be ideal. The code inside the if check is currently only used to initialise the marker with the correct tilt value not to update it every time (this is done through a setter).